rst_files/aiyagari.rst

```diff
@@ -227,64 +198,46 @@ Here we take the state to be :math:`s_t := (a_t, z_t)`, where :math:`a_t` is ass
 
 The action is the choice of next period asset level :math:`a_{t+1}`
 
+The type also includes a default set of parameters that we'll adopt unless otherwise specified
 
+Setup
+-----
 
-
-
-The type also includes a default set of parameters that we'll adopt unless otherwise specified
+.. literalinclude:: /_static/includes/deps.jl
 
 .. code-block:: julia
-  :class: test
+    :class: test
 
-  using Test
+    using Test, Random
 
 .. code-block:: julia
 
-    using QuantEcon, Random, LinearAlgebra
-
-    mutable struct Household{TR<:Real, TF<:AbstractFloat, TI<:Integer, VF <: AbstractVector{TF}}
-        r::TR
-        w::TR
-        σ::TR
-        β::TF
-        z_chain::MarkovChain{TF, Matrix{TF}, Vector{TF}}
-        a_min::TR
-        a_max::TR
-        a_size::TI
-        a_vals::VF
-        z_size::TI
-        n::TI
-        s_vals::Matrix{TF}
-        s_i_vals::Matrix{TI}
-        R::Matrix{TR}
-        Q::Array{TR,3}
-        u::Function
-    end
+    using Parameters, Plots, QuantEcon
+    gr(fmt = :png)
 
+.. code-block:: julia
 
-    function Household(;
-                        r=0.01,
-                        w=1.0,
-                        σ=1.0,
-                        β=0.96,
-                        z_chain
-                            =MarkovChain([0.9 0.1; 0.1 0.9], [0.1; 1.0]),
-                        a_min=1e-10,
-                        a_max=18.0,
-                        a_size=200)
-
-        # set up grids
-        a_vals = range(a_min, a_max, length = a_size)
-        z_size = length(z_chain.state_values)
-        n = a_size*z_size
-        s_vals = gridmake(a_vals, z_chain.state_values)
-        s_i_vals = gridmake(1:a_size, 1:z_size)
-
-        # set up Q
-        Q = zeros(n, a_size, n)
-        for next_s_i in 1:n
-            for a_i in 1:a_size
-                for s_i in 1:n
+    Household = @with_kw (r = 0.01,
+                          w = 1.0,
+                          σ = 1.0,
+                          β = 0.96,
+                          z_chain = MarkovChain([0.9 0.1; 0.1 0.9], [0.1; 1.0]),
+                          a_min = 1e-10,
+                          a_max = 18.0,
+                          a_size = 200,
+                          a_vals = range(a_min, a_max, length = a_size),
+                          z_size = length(z_chain.state_values),
+                          n = a_size * z_size,
+                          s_vals = gridmake(a_vals, z_chain.state_values),
+                          s_i_vals = gridmake(1:a_size, 1:z_size),
+                          u = σ == 1 ? x -> log(x) : x -> (x^(1 - σ) - 1) / (1 - σ),
+                          R = setup_R!(fill(-Inf, n, a_size), a_vals, s_vals, r, w, u),
```

Comment:
Can the `-Inf` here be replaced with `0`?

Reply:
The `-Inf` is suppose to represent a really really bad utility (no consumption, then the agent dies). It follows the math. Zero doesn't carry that connotation.

Reply:
Gotcha, thanks. I'll throw in a comment about it.